metpy/calc/thermo.py

```diff
@@ -185,22 +186,15 @@ def lcl(pressure, temperature, dewpt, max_iters=50, eps=1e-2):
     The function is guaranteed to finish by virtue of the `max_iters` counter.
 
     """
-    w = mixing_ratio(saturation_vapor_pressure(dewpt), pressure)
-    new_p = p = pressure
-    eps = units.Quantity(eps, p.units)
-    while max_iters:
-        td = dewpoint(vapor_pressure(p, w))
-        new_p = pressure * (td / temperature) ** (1. / kappa)
-        if np.abs(new_p - p).max() < eps:
-            break
-        p = new_p
-        max_iters -= 1
-
-    else:
-        # We have not converged
-        raise RuntimeError('LCL calculation has not converged.')
+    def _lcl_iter(p, p0, w, t):
+        td = dewpoint(vapor_pressure(units.Quantity(p, pressure.units), w))
+        return (p0 * (td / t) ** (1. / kappa)).m
 
-    return new_p, td
+    w = mixing_ratio(saturation_vapor_pressure(dewpt), pressure)
+    fp = so.fixed_point(_lcl_iter, pressure.m, args=(pressure.m, w, temperature),
+                        xtol=eps, maxiter=max_iters, method='del2')
```

Comment:
This would also fix the break on legacy where the method argument is not supported.